### PR TITLE
test: Unit test for src/utill.rs

### LINKED ISSUE #125: test: Unit test for src/utill.rs
Issue Description:
This issue highlights the need for unit tests in src/utill.rs file. Currently, some functions in this file are untested, which could lead to potential issues in the future.

Objective:
Write unit tests for the untested functions in the src/utill.rs file.

Details:

Cover all  functions with appropriate test cases.
Ensure tests validate functionality, handle edge cases, and check error conditions.
Integrate the tests into our existing test suite.

### PR DESCRIPTION
**Pull request by Jem256**
Fixes #125

### CODE REVIEW

File: src/utill.rs [hunk 1]
```diff
@@ -526,6 +526,30 @@ mod tests {
         );
     }
 
+    #[test]
```
Comment (Shourya742) [inline src/utill.rs:529]: @Jem256 You can incorporate all these failure cases into a single test method.

**Review by Shourya742 — COMMENTED** [review]
Concept ACK

File: src/utill.rs (hunk 1 above)
Comment (Jem256) [reply src/utill.rs:529]: > @Jem256 You can incorporate all these failure cases into a single test method.

I have done that.

**Review by Shourya742 — CHANGES_REQUESTED** [review]
Excellent work! Could you please cover the remaining misses in the same manner?

**Review by Shourya742 — APPROVED** [review]
ACK [5045fb8](https://github.com/utxo-teleport/teleport-transactions/pull/138/commits/5045fb88194172ccc1a6f02d2304ca8ab3c63f0c)

**Review by rajarshimaitra — APPROVED** [review]
ACK
